static/format-sdb.sh

```diff
@@ -22,21 +22,34 @@ mkdir -p "$MOUNT_"
 
 
 # Check what Hypervisor disks are available
-if partprobe /dev/sdb &>/dev/null;	#HyperV, VMware, VirtualBox Hypervisors
+SYSVENDOR=$(cat /sys/devices/virtual/dmi/id/sys_vendor)
+if [ "$SYSVENDOR" == "VMware, Inc." ];
 then
+    SYSNAME="VMware"
     DEVTYPE=sdb
-elif partprobe /dev/xvdb &>/dev/null;	#Xen Hypervisors
+elif [ "$SYSVENDOR" == "Microsoft Corporation" ];
 then
+    SYSNAME="Hyper-V"
+    DEVTYPE=sdb
+elif [ "$SYSVENDOR" == "innotek GmbH" ];
+then
+    SYSNAME="VirtualBox"
+    DEVTYPE=sdb
+elif [ "$SYSVENDOR" == "Xen" ];
+then
+    SYSNAME="Xen/XCP-NG"
     DEVTYPE=xvdb
-elif partprobe /dev/vdb &>/dev/null;	#KVM Hypervisor
+elif [ "$SYSVENDOR" == "QEMU" ];
 then
+    SYSNAME="KVM/QEMU"
     DEVTYPE=vdb
+elif [ "$SYSVENDOR" == "DigitalOcean" ];
+then
+    SYSNAME="DigitalOcean"
+    DEVTYPE=sda
 else
-msg_box "It seems like you didn't mount a second disk. 
-To be able to put the DATA on a second drive formatted as ZFS you need to add a second disk to this server.
-
-This script will now exit. Please mount a second disk and start over."
-exit 1
```

Comment:
It does it twice, once here and then again at Line 98, if it fails here then line 98 is no longer required, unless I'm reading it wrong.
https://github.com/nextcloud/vm/blob/4ebd80dbfbb1b5b061c39a417786c42af07f8aa3/static/format-sdb.sh#L98

The second reason for removing it here and not the later one is that because I am now checking the Vendors we need an else statement if they try to run it on a hypervisor that we don't pull the vendor name for... or they use a physical machine, which could pull various Vendor names.

Reply:
As you also can see it installs ZFS dependencies directly after, here: https://github.com/nextcloud/vm/blob/e784974f81e745f4bb5ed51ea566eed7508df11b/static/format-sdb.sh#L56

I added the check and `exit 1` so that nothing is installed or changed if the user didn't create a second drive.

I do think that the message could be more clear, and also make it an option like:
1. Disk mounted = "continue"
2. Disk NOT mounted = Tell the user that the disk isn't mounted and ask if the script should continue anyway, resulting in that /mnt/ncdata will hold the DATA but not on a second drive as intended.

What do you think?

Btw, please also change the URL to hanssonit.se here: https://github.com/nextcloud/vm/blob/e784974f81e745f4bb5ed51ea566eed7508df11b/static/format-sdb.sh#L139

Reply:
Right yes that makes sense, I will make some amendments and see what I can do, although I don't think installing the zfs utilties before it fails is really a concern. I only say this because and I don't know if it's just me, when testing I'll forget to attach the second drive.

So I run the installation script which is fine, I exit at this part and mount the second drive, but now cannot run the nextcloud_install_production.sh as it has already installed apache and says my server is not a clean installation, so I end up reinstalling Ubuntu anyway.

Maybe we should look into moving the format-sdb.sh closer to the start of the script?

Reply:
Yeah, ok, let's move it before Apache installation.